utils/collection/stack/interface.go

```diff
@@ -0,0 +1,26 @@
+package stack
+
+import "iter"
+
+//go:generate go tool mockgen -destination=../../mocks/mock_$GOPACKAGE.go -package=mocks github.com/ARM-software/golang-utils/utils/collection/$GOPACKAGE IStack
+
+// IStack specifies the behaviour of a last-in, first-out (LIFO) collection.
+// it is inspired by the work of https://github.com/hayageek/threadsafe/ and https://github.com/golang-collections/collections
+type IStack[T any] interface {
+	// Push adds elements to the stack.
+	Push(value ...T)
+	// PushSequence adds elements to the stack.
+	PushSequence(seq iter.Seq[T])
+	// Pop removes and returns an element from the stack.
+	Pop() T
+	// Peek returns the element at the top of the stack without removing it.
+	Peek() T
```

Comment:
```suggestion
	Peek() (T, bool)
```

I think these suplement it rather than constantly call IsEmpty

Reply:
the reason I made it as it was is because it is quite close to how channel work (with regards to queues)

Reply:
Then enforce that these will block if there is nothing in them.

But channels have so many footguns that I wouldn't use them as inspiration for a high-level container